jlm/mlir/frontend/MlirToJlmConverter.cpp

```diff
@@ -0,0 +1,244 @@
+  std::unordered_map<::mlir::Operation *, rvsdg::node *> operations;
+  for (auto & mlirOp : block.getOperations())
+  {
+    std::vector<const jlm::rvsdg::output *> inputs;
+    for (auto operand : mlirOp.getOperands())
+    {
+      if (auto * producer = operand.getDefiningOp())
+      {
+        size_t index = GetOperandIndex(producer, operand);
+        inputs.push_back(operations[producer]->output(index));
+      }
+      else
+      {
+        // If there is no defining op, the Value is necessarily a Block argument.
+        JLM_ASSERT(::mlir::dyn_cast<::mlir::BlockArgument>(operand));
+        inputs.push_back(
+            rvsdgRegion.argument(operand.cast<::mlir::BlockArgument>().getArgNumber()));
+      }
+    }
+
+    if (auto * node = ConvertOperation(mlirOp, rvsdgRegion, inputs))
+    {
+      operations[&mlirOp] = node;
+    }
+  }
+
+  // The results of the region/block are encoded in the terminator operation
+  auto terminator = block.getTerminator();
+  std::vector<jlm::rvsdg::output *> results;
+  for (auto operand : terminator->getOperands())
+  {
+    if (auto * producer = operand.getDefiningOp())
+    {
+      size_t index = GetOperandIndex(producer, operand);
+      results.push_back(operations[producer]->output(index));
+    }
+    else
+    {
+      // If there is no defining op, the Value is necessarily a Block argument.
+      JLM_ASSERT(::mlir::dyn_cast<::mlir::BlockArgument>(operand));
+      results.push_back(rvsdgRegion.argument(operand.cast<::mlir::BlockArgument>().getArgNumber()));
+    }
+  }
+
+  return results;
+}
+
+rvsdg::node *
+MlirToJlmConverter::ConvertOperation(
+    ::mlir::Operation & mlirOperation,
+    rvsdg::region & rvsdgRegion,
+    std::vector<const rvsdg::output *> & inputs)
+{
+  if (::mlir::isa<::mlir::rvsdg::OmegaNode>(&mlirOperation))
+  {
+    ConvertOmega(mlirOperation, rvsdgRegion);
+    // Omega doesn't have a corresponding RVSDG node so we return nullptr
+    return nullptr;
+  }
+  else if (::mlir::isa<::mlir::rvsdg::LambdaNode>(&mlirOperation))
+  {
+    return ConvertLambda(mlirOperation, rvsdgRegion);
+  }
+  else if (auto constant = ::mlir::dyn_cast<::mlir::arith::ConstantIntOp>(&mlirOperation))
+  {
+    auto type = constant.getType();
+    JLM_ASSERT(type.getTypeID() == ::mlir::IntegerType::getTypeID());
+    auto integerType = ::mlir::cast<::mlir::IntegerType>(type);
+
+    return rvsdg::node_output::node(
+        rvsdg::create_bitconstant(&rvsdgRegion, integerType.getWidth(), constant.value()));
+  }
+  else if (
+      ::mlir::isa<::mlir::rvsdg::LambdaResult>(&mlirOperation)
+      || ::mlir::isa<::mlir::rvsdg::OmegaResult>(&mlirOperation))
+  {
+    // This is a terminating operation that doesn't have a corresponding RVSDG node
+    return nullptr;
+  }
+  else
+  {
+    JLM_UNREACHABLE(
+        util::strfmt(
+            "Operation not implemented:" + mlirOperation.getName().getStringRef().str() + "\n")
+            .c_str());
```

Comment:
```
auto message = util::strfmt("Operation not implemented: ", mlirOperation.getName().getStringRef().str(), "\n");
JLM_UNREACHABLE(message.c_str());
```
Note the commas in `strfmt()` instead of the plusses. This is not a big issue, but since I had to comment on the if-statement issue, I also thought I can comment on this one.

Reply:
Fixed